src/reducers/appsReducer.ts

```diff
@@ -8,98 +8,94 @@ import { Reducer } from 'redux'
-                // Since we're updating training status simulate update to datetime field
-                datetime: new Date(),
+
+            app.didPollingExpire = false
+            app.trainingStatus = action.trainingStatus.trainingStatus
+            // Since we're updating training status simulate update to datetime field
+            app.datetime = new Date(),
                 // Used discriminated union to access failure message
-                trainingFailureMessage: (action.trainingStatus.trainingStatus === TrainingStatusCode.Failed)
+                app.trainingFailureMessage = (action.trainingStatus.trainingStatus === TrainingStatusCode.Failed)
                     ? action.trainingStatus.trainingFailureMessage
                     : null
-            }
-
-            return { ...state, all: replace(state.all, newApp, a => a.appId) }
+            return
         }
         case AT.CREATE_APPLICATION_FULFILLED:
-            return { ...state, all: [...state.all, action.app], selectedAppId: action.app.appId }
+            state.all.push(action.app)
+            state.selectedAppId = action.app.appId
+            return
         case AT.SET_CURRENT_APP_FULFILLED:
-            return { ...state, selectedAppId: action.app.appId };
+            state.selectedAppId = action.app.appId
+            return
         case AT.DELETE_APPLICATION_FULFILLED:
-            return { ...state, all: state.all.filter(app => app.appId !== action.appId), selectedAppId: undefined };
+            state.all = state.all.filter(app => app.appId !== action.appId)
+            state.selectedAppId = undefined
+            return
         case AT.EDIT_APPLICATION_FULFILLED:
         case AT.CREATE_APP_TAG_FULFILLED:
         case AT.EDIT_APP_LIVE_TAG_FULFILLED:
-            return { ...state, all: replace(state.all, action.app, app => app.appId) }
+            state.all = replace(state.all, action.app, app => app.appId)
+            return
         case AT.EDIT_APP_EDITING_TAG_FULFILLED:
-            return { ...state, activeApps: action.activeApps }
+            state.activeApps = action.activeApps
+            return
 
         // TODO: We're expecting more handlers here, as we're simply updating lastModifiedDateTime...
         case AT.EDIT_TRAINDIALOG_FULFILLED:
 
-            const app = state.all.find(a => a.appId === action.appId)
-
-            // App may have been deleted
+            const app = state.all.find(a => a.appId === action.appId) as App | undefined
+            // User may have delete the app
             if (!app) {
-                return state;
-            }
-
-            const newApp: App = {
-                ...(app as App),
-                lastModifiedDateTime: `${new Date().toISOString().slice(0, 19)}+00:00`,
+                return
             }
 
-            return { ...state, all: replace(state.all, newApp, a => a.appId) }
-
+            app.lastModifiedDateTime = `${new Date().toISOString().slice(0, 19)}+00:00`
```

Comment:
new functionality

Reply:
What's new?

Previously it was assigning to the `newApp` object. Now it assigns to the original `app` but it was always updating the time.

Btws, why do we force the offset to be 0 like `.slice(0, 19)}+00:00`?
To me it seems misleading as that's not actually when the time was updated.

Reply:
Responding again to force notification